File: CS/MyMvpvmApplication/ViewModels/TaskCollectionViewModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using MyMvpvmApplication.Model;
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using DevExpress.Mvvm.DataAnnotations;

namespace MyMvpvmApplication.ViewModels {
    public class TaskCollectionViewModel {
        IList<Task> tasksCore;

        public TaskCollectionViewModel() {
            tasksCore = new BindingList<Task> { 
                new Task(){ Subject="Test Task", Description = "Test Description"}
            };
        }
        public IList<Task> Tasks {
            get { return tasksCore; }
        }

        public void AddTask() {
            Tasks.Add(new Task() { Subject = "Task " + Tasks.Count.ToString() });
        }

        public virtual Task SelectedTask {
            get;
            set;
        }
        protected virtual void OnSelectedTaskChanged() {
            this.RaiseCanExecuteChanged(x => x.DeleteTask(null));
            Messenger.Default.Send(new TaskMessage() { Task = SelectedTask });
        }

        [Command(UseCommandManager = false)]
        public void DeleteTask(Task task) {
            Tasks.Remove(task);
        }
        public bool CanDeleteTask(Task task) {
            return task != null;
        }

    }
}

File: VB/ViewModels/TaskCollectionViewModel.vb

Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Text
Imports MyMvpvmApplication.Model
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Mvvm.DataAnnotations

Namespace MyMvpvmApplication.ViewModels
    Public Class TaskCollectionViewModel
        Private tasksCore As IList(Of Task)

        Public Sub New()
            tasksCore = New BindingList(Of Task)() From { _
                New Task() With { _
                    .Subject = "Test Task", _
                    .Description = "Test Description" _
                } _
            }
        End Sub
        Public ReadOnly Property Tasks() As IList(Of Task)
            Get
                Return tasksCore
            End Get
        End Property

        Public Sub AddTask()
            Tasks.Add(New Task() With { _
                .Subject = "Task " + Tasks.Count.ToString() _
            })
        End Sub

        Public Overridable Property SelectedTask() As Task

        Protected Overridable Sub OnSelectedTaskChanged()
            Me.RaiseCanExecuteChanged(Sub(x) x.DeleteTask(Nothing))
            Messenger.[Default].Send(New TaskMessage() With { _
                .Task = SelectedTask _
            })
        End Sub

        <Command(UseCommandManager:=False)> _
        Public Sub DeleteTask(task As Task)
            Tasks.Remove(task)
        End Sub
        Public Function CanDeleteTask(task As Task) As Boolean
            Return task IsNot Nothing
        End Function

    End Class
End Namespace